hs_file_types/utils.py

```diff
@@ -396,3 +431,30 @@ def set_logical_file_type(res, user, file_id, hs_file_type=None, folder_path='',
         if fail_feedback:
             raise
         return None
+
+
+def get_logical_file_metadata_json_schema(file_with_path):
+    """
+    Get logical file metadata values in json format along with its json schema
+    for metadata display
+    :param file_with_path: meatadata file with path
+    :return: metadata JSON values and schema for logical file
+    """
+    istorage = IrodsStorage()
+    with istorage.open(file_with_path) as f:
+        metadata = load_rdf(f.read())
+        json_value = metadata.json()
+        json_schema = metadata.schema_json()
```

Comment:
@sblack-usu Two quick questions for your input. The answers to both questions are most likely yes, but want to get your confirmation/input just in case.
- Should I check metadata update flag and update metadata xml file if needed before loading it to rdf metadata model due to our lazy metadata update implementation? 
- Since model program and model instance aggregations are not yet implemented in hsmodels, should I add a check here to use old way for metadata view/update for model program and model instance aggregations? If so, the front end code needs to be kept as well for these two model aggregations, which means we will have to keep front end code for both the old and new ways in order to accommodate this case.

Reply:
The aggregation metadata isn't (shouldn't?) be lazy.  My understanding is that those files are always kept up to date.  We should verify that is true.

We will have to manage both systems until model program/instance is implemented in hsmodels.  I am currently working on the json API endpoints using hsmodels, so hopefully I can work out some of the other blockers you've experienced.